opnode/cmd/main.go

```diff
@@ -1,31 +1,211 @@
 package main
 
 import (
+	"context"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/params"
+	"github.com/urfave/cli"
+
 	"github.com/ethereum-optimism/optimistic-specs/opnode/node"
-	"github.com/protolambda/ask"
 )
 
-type MainCmd struct {
+var (
+	Version   = ""
+	GitCommit = ""
+	GitDate   = ""
+)
+
+func main() {
+	// Set up logger with a default INFO level in case we fail to parse flags,
+	// otherwise the final critical log won't show what the parsing error was.
+	log.Root().SetHandler(
+		log.LvlFilterHandler(
+			log.LvlInfo,
+			log.StreamHandler(os.Stdout, log.TerminalFormat(true)),
+		),
+	)
+
+	app := cli.NewApp()
+	app.Flags = Flags
+	app.Version = fmt.Sprintf("%s-%s", Version, params.VersionWithCommit(GitCommit, GitDate)) // TODO: Don't use Geth's hardcoded versions
```

Comment:
Can we just re-implement `VersionWithCommit` (and maybe simplify it) to not default to the geth hardcoded version? I would be fine with a `dev` placeholder version string for now.

Reply:
I went with a placeholder. This is another thing from the l2os (I left a comment there about it)